src/gmail.js

```diff
@@ -1870,8 +1877,22 @@ var Gmail = function(localJQuery) {
 
         for (let email of email_data) {
             // cache email directly on IDs
-            c.emailIdCache[email.id] = email;
-            c.emailLegacyIdCache[email.legacy_email_id] = email;
+            if (c.emailIdCache[email.id] === undefined) {
+                //console.log("ADD c.emailIdCache[email.id]",c.emailIdCache[email.id],email);
+                c.emailIdCache[email.id] = email;
+            }
+            else if (data_source === "fd_request_payload" || data_source === "fd_embedded_json") {
```

Comment:
If these two sources are the only sources which are going to be acted on, wouldn't it be better to just combine them into a boolean, like `isAuthorativeSource` or something?

I think it might make for fewer errors when calling this function, and easier to understand the semantic meaning of the parameter.

Reply:
I agree, I combined the 2 data source with a boolean.
Code is clearer.